cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSICKANSink.java

```diff
@@ -409,29 +411,35 @@ public String buildOrgName(String fiwareService) throws CygnusBadConfiguration {
      * @return Package name
      * @throws CygnusBadConfiguration
      */
-    public String buildPkgName(String fiwareService, String fiwareServicePath) throws CygnusBadConfiguration {
+    public String buildPkgName(String fiwareService, String fiwareServicePath, String entityId) throws CygnusBadConfiguration {
         String pkgName;
         
-        if (enableEncoding) {
-            pkgName = NGSICharsets.encodeCKAN(fiwareService)
-                    + CommonConstants.CONCATENATOR
-                    + NGSICharsets.encodeCKAN(fiwareServicePath);
-        } else {
-            if (fiwareServicePath.equals("/")) {
-                pkgName = NGSIUtils.encode(fiwareService, false, true).toLowerCase(Locale.ENGLISH);
-            } else {
-                pkgName = (NGSIUtils.encode(fiwareService, false, true)
-                        + NGSIUtils.encode(fiwareServicePath, false, true)).toLowerCase(Locale.ENGLISH);
-            } // if else
-        } // if else
-
-        if (pkgName.length() > NGSIConstants.CKAN_MAX_NAME_LEN) {
-            throw new CygnusBadConfiguration("Building package name '" + pkgName + "' and its length is "
-                    + "greater than " + NGSIConstants.CKAN_MAX_NAME_LEN);
-        } else if (pkgName.length() < NGSIConstants.CKAN_MIN_NAME_LEN) {
-            throw new CygnusBadConfiguration("Building package name '" + pkgName + "' and its length is "
-                    + "lower than " + NGSIConstants.CKAN_MIN_NAME_LEN);
-        } // if else if
+        switch(dataModel) {
```

Comment:
As far as I understand (according to https://github.com/telefonicaid/fiware-cygnus/pull/1850/files#diff-7d759ac06f173f8e331ae811fbf231c7R354) CKAN sink support two modes: dm-by-entity-id and dm-by-entity. I see the following pattern in this PR (in several places):

```
        switch(dataModel) {
            case DMBYENTITYID:
                // Code for processing the dm-by-entity-id case
                break;
            default:
                // Code for processing the dm-by-entity
                break;
```

However, I'd suggest a more extensible and robuts approach:

```
        switch(dataModel) {
            case DMBYENTITYID:
                // Code for processing the dm-by-entity-id case
                break;
            case DMBYENTITY:
                // Code for processing the dm-by-entity
                break;
            default:
                // Deal with the error of trying to use an unsupported mode (log, etc.)
                break;
```

Reply:
Fixed in [1651f0d](https://github.com/telefonicaid/fiware-cygnus/pull/1850/commits/1651f0dbeb116b8222d7fe9df90bd4d603b17b5e)